src/client_backend/triton_c_api/triton_loader.cc

```diff
@@ -244,16 +244,25 @@ TritonLoader::StartTriton()
   REPORT_TRITONSERVER_ERROR(
       api_version_fn_(&api_version_major, &api_version_minor),
       "unable to get api version");
-  if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
-      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+
+  if (TRITONSERVER_API_VERSION_MAJOR != api_version_major) {
```

Comment:
Can we do something like this?

```
std::string createErrorMessage(const std::string& message, int expected, int actual) {
    return message + "Expected version: " + std::to_string(expected) + "\n" +
           "Actual version: " + std::to_string(actual) + "\n";
}

if (TRITONSERVER_API_VERSION_MAJOR != api_version_major) {
    std::string errorMessage = createErrorMessage(
        "Triton server API major version mismatch.\n",
        TRITONSERVER_API_VERSION_MAJOR, api_version_major
    );
    return Error(errorMessage);
}

if (TRITONSERVER_API_VERSION_MINOR > api_version_minor) {
    std::string warningMessage = createErrorMessage(
        "Warning: Triton server API minor version mismatch.\n",
        TRITONSERVER_API_VERSION_MINOR, api_version_minor
    );
    std::cerr << warningMessage;
}

```

The createErrorMessage function avoids redundant code by consolidating string message construction.
Use of std::string: Instead of std::stringstream, std::string is used with std::to_string for simplicity and efficiency.
Reduced redundancy by eliminating the need for separate std::stringstream instances.

Reply:
I don't have an idea which is better, I used existing. I will consider this, research a bit once and make changes as you suggested.